pkg/github/pagination.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2024 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package github
+
+import (
+	"fmt"
+
+	"github.com/google/go-github/v61/github"
+)
+
+func Paginate(f func(opts *github.ListOptions) (*github.Response, error)) error {
```

Comment:
Needs docs if people agree on approach

Reply:
Don't feel it's super reusable outside. Maybe keep it private for now.

Reply:
done, but added docs.